From 95822831997e71319bf7f1e2cf44f9fcc9116fe5 Mon Sep 17 00:00:00 2001
From: Michael Heimpold <michael.heimpold@in-tech.com>
Date: Mon, 17 Feb 2020 12:02:24 +0100
Subject: [PATCH] Partly revert "meson: stop creating enablement symlinks in
 /etc  during installation"

This partly reverts commit 01d2041e41f4886a6dff16a53a950ae8d5b66bc7.

Signed-off-by: Michael Heimpold <michael.heimpold@in-tech.com>
---
 units/meson.build | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/units/meson.build b/units/meson.build
index 6a3a0d0dea..210ebd4931 100644
--- a/units/meson.build
+++ b/units/meson.build
@@ -94,7 +93,8 @@ units = [
          'sockets.target.wants/'],
         ['systemd-journald.socket',             '',
          'sockets.target.wants/'],
-        ['systemd-networkd.socket',             'ENABLE_NETWORKD'],
+        ['systemd-networkd.socket',             'ENABLE_NETWORKD',
+         join_paths(pkgsysconfdir, 'system/sockets.target.wants/')],
         ['systemd-poweroff.service',             ''],
         ['systemd-reboot.service',               ''],
         ['systemd-rfkill.socket',               'ENABLE_RFKILL'],
@@ -175,8 +174,11 @@ in_units = [
         ['systemd-modules-load.service',         'HAVE_KMOD',
          'sysinit.target.wants/'],
         ['systemd-network-generator.service',    'ENABLE_NETWORKD'],
-        ['systemd-networkd.service',             'ENABLE_NETWORKD'],
-        ['systemd-networkd-wait-online.service', 'ENABLE_NETWORKD'],
+        ['systemd-networkd.service',             'ENABLE_NETWORKD',
+         join_paths(pkgsysconfdir, 'system/dbus-org.freedesktop.network1.service') + ' ' +
+         join_paths(pkgsysconfdir, 'system/multi-user.target.wants/')],
+        ['systemd-networkd-wait-online.service', 'ENABLE_NETWORKD',
+         join_paths(pkgsysconfdir, 'system/network-online.target.wants/')],
         ['systemd-nspawn@.service',              ''],
         ['systemd-portabled.service',            'ENABLE_PORTABLED',
          'dbus-org.freedesktop.portable1.service'],
@@ -184,7 +186,9 @@ in_units = [
         ['systemd-random-seed.service',          'ENABLE_RANDOMSEED',
          'sysinit.target.wants/'],
         ['systemd-remount-fs.service',           ''],
-        ['systemd-resolved.service',             'ENABLE_RESOLVE'],
+        ['systemd-resolved.service',             'ENABLE_RESOLVE',
+         join_paths(pkgsysconfdir, 'system/dbus-org.freedesktop.resolve1.service') + ' ' +
+         join_paths(pkgsysconfdir, 'system/multi-user.target.wants/')],
         ['systemd-rfkill.service',               'ENABLE_RFKILL'],
         ['systemd-suspend.service',              ''],
         ['systemd-sysctl.service',               '',
@@ -193,7 +197,8 @@ in_units = [
          'sysinit.target.wants/'],
         ['systemd-timedated.service',            'ENABLE_TIMEDATED',
          'dbus-org.freedesktop.timedate1.service'],
-        ['systemd-timesyncd.service',            'ENABLE_TIMESYNCD'],
+        ['systemd-timesyncd.service',            'ENABLE_TIMESYNCD',
+         join_paths(pkgsysconfdir, 'system/sysinit.target.wants/')],
         ['systemd-time-wait-sync.service',       'ENABLE_TIMESYNCD'],
         ['systemd-tmpfiles-clean.service',       'ENABLE_TMPFILES'],
         ['systemd-tmpfiles-setup-dev.service',   'ENABLE_TMPFILES',
-- 
2.17.1

